<?xml version='1.0' encoding='UTF-8'?>
<root>
 <tabbed_widget parent="main_window" name="Main Window">
  <Tab containers="1" tab_name="Position">
   <Container>
    <DockSplitter count="2" orientation="-" sizes="0.5;0.5">
     <DockSplitter count="3" orientation="|" sizes="0.333538;0.332923;0.333538">
      <DockArea name="...">
       <plot flip_y="false" flip_x="false" style="Lines" mode="TimeSeries">
        <range right="1656161137.982903" bottom="0.007380" top="0.007380" left="1656161122.983859"/>
        <limitY/>
        <curve color="#1f77b4" name="/iiwa_front/adrc_trajectory_controller/state/desired/positions.0"/>
        <curve color="#d62728" name="/iiwa_front/adrc_trajectory_controller/state/actual/positions.0"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot flip_y="false" flip_x="false" style="Lines" mode="TimeSeries">
        <range right="1656161137.982903" bottom="-0.043725" top="-0.043724" left="1656161122.983859"/>
        <limitY/>
        <curve color="#1f77b4" name="/iiwa_front/adrc_trajectory_controller/state/desired/positions.1"/>
        <curve color="#d62728" name="/iiwa_front/adrc_trajectory_controller/state/actual/positions.1"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot flip_y="false" flip_x="false" style="Lines" mode="TimeSeries">
        <range right="1656161137.982903" bottom="0.007572" top="0.007572" left="1656161122.983859"/>
        <limitY/>
        <curve color="#1f77b4" name="/iiwa_front/adrc_trajectory_controller/state/desired/positions.2"/>
        <curve color="#d62728" name="/iiwa_front/adrc_trajectory_controller/state/actual/positions.2"/>
       </plot>
      </DockArea>
     </DockSplitter>
     <DockSplitter count="4" orientation="|" sizes="0.25;0.25;0.25;0.25">
      <DockArea name="...">
       <plot flip_y="false" flip_x="false" style="Lines" mode="TimeSeries">
        <range right="1656161137.982903" bottom="-1.116526" top="-1.116520" left="1656161122.983859"/>
        <limitY/>
        <curve color="#1f77b4" name="/iiwa_front/adrc_trajectory_controller/state/desired/positions.3"/>
        <curve color="#d62728" name="/iiwa_front/adrc_trajectory_controller/state/actual/positions.3"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot flip_y="false" flip_x="false" style="Lines" mode="TimeSeries">
        <range right="1656161137.982903" bottom="-0.005421" top="-0.005420" left="1656161122.983859"/>
        <limitY/>
        <curve color="#1f77b4" name="/iiwa_front/adrc_trajectory_controller/state/desired/positions.4"/>
        <curve color="#d62728" name="/iiwa_front/adrc_trajectory_controller/state/actual/positions.4"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot flip_y="false" flip_x="false" style="Lines" mode="TimeSeries">
        <range right="1656161137.982903" bottom="1.677869" top="1.677870" left="1656161122.983859"/>
        <limitY/>
        <curve color="#1f77b4" name="/iiwa_front/adrc_trajectory_controller/state/desired/positions.5"/>
        <curve color="#d62728" name="/iiwa_front/adrc_trajectory_controller/state/actual/positions.5"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot flip_y="false" flip_x="false" style="Lines" mode="TimeSeries">
        <range right="1656161137.982903" bottom="0.013364" top="0.013365" left="1656161122.983859"/>
        <limitY/>
        <curve color="#1f77b4" name="/iiwa_front/adrc_trajectory_controller/state/desired/positions.6"/>
        <curve color="#d62728" name="/iiwa_front/adrc_trajectory_controller/state/actual/positions.6"/>
       </plot>
      </DockArea>
     </DockSplitter>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab containers="1" tab_name="Velocities">
   <Container>
    <DockSplitter count="2" orientation="-" sizes="0.5;0.5">
     <DockSplitter count="3" orientation="|" sizes="0.333538;0.332923;0.333538">
      <DockArea name="...">
       <plot flip_y="false" flip_x="false" style="Lines" mode="TimeSeries">
        <range right="1656161136.598776" bottom="-0.000066" top="0.000068" left="1656161121.598993"/>
        <limitY/>
        <curve color="#1f77b4" name="/iiwa_front/adrc_trajectory_controller/state/desired/velocities.0"/>
        <curve color="#d62728" name="/iiwa_front/adrc_trajectory_controller/state/actual/velocities.0"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot flip_y="false" flip_x="false" style="Lines" mode="TimeSeries">
        <range right="1656161136.598776" bottom="-0.000049" top="0.000057" left="1656161121.598993"/>
        <limitY/>
        <curve color="#1f77b4" name="/iiwa_front/adrc_trajectory_controller/state/desired/velocities.1"/>
        <curve color="#d62728" name="/iiwa_front/adrc_trajectory_controller/state/actual/velocities.1"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot flip_y="false" flip_x="false" style="Lines" mode="TimeSeries">
        <range right="1656161136.598776" bottom="-0.000057" top="0.000058" left="1656161121.598993"/>
        <limitY/>
        <curve color="#1f77b4" name="/iiwa_front/adrc_trajectory_controller/state/desired/velocities.2"/>
        <curve color="#d62728" name="/iiwa_front/adrc_trajectory_controller/state/actual/velocities.2"/>
       </plot>
      </DockArea>
     </DockSplitter>
     <DockSplitter count="4" orientation="|" sizes="0.25;0.25;0.25;0.25">
      <DockArea name="...">
       <plot flip_y="false" flip_x="false" style="Lines" mode="TimeSeries">
        <range right="1656161136.598776" bottom="-0.000223" top="0.000269" left="1656161121.598993"/>
        <limitY/>
        <curve color="#1f77b4" name="/iiwa_front/adrc_trajectory_controller/state/desired/velocities.3"/>
        <curve color="#d62728" name="/iiwa_front/adrc_trajectory_controller/state/actual/velocities.3"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot flip_y="false" flip_x="false" style="Lines" mode="TimeSeries">
        <range right="1656161136.598776" bottom="-0.000139" top="0.000133" left="1656161121.598993"/>
        <limitY/>
        <curve color="#1f77b4" name="/iiwa_front/adrc_trajectory_controller/state/desired/velocities.4"/>
        <curve color="#d62728" name="/iiwa_front/adrc_trajectory_controller/state/actual/velocities.4"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot flip_y="false" flip_x="false" style="Lines" mode="TimeSeries">
        <range right="1656161136.598776" bottom="-0.000154" top="0.000150" left="1656161121.598993"/>
        <limitY/>
        <curve color="#1f77b4" name="/iiwa_front/adrc_trajectory_controller/state/desired/velocities.5"/>
        <curve color="#d62728" name="/iiwa_front/adrc_trajectory_controller/state/actual/velocities.5"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot flip_y="false" flip_x="false" style="Lines" mode="TimeSeries">
        <range right="1656161136.598776" bottom="-0.000114" top="0.000104" left="1656161121.598993"/>
        <limitY/>
        <curve color="#1f77b4" name="/iiwa_front/adrc_trajectory_controller/state/desired/velocities.6"/>
        <curve color="#d62728" name="/iiwa_front/adrc_trajectory_controller/state/actual/velocities.6"/>
       </plot>
      </DockArea>
     </DockSplitter>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab containers="1" tab_name="Effort">
   <Container>
    <DockSplitter count="2" orientation="-" sizes="0.5;0.5">
     <DockSplitter count="3" orientation="|" sizes="0.333538;0.332923;0.333538">
      <DockArea name="...">
       <plot flip_y="false" flip_x="false" style="Lines" mode="TimeSeries">
        <range right="1656161145.674849" bottom="0.193479" top="0.277197" left="1656161130.674926"/>
        <limitY/>
        <curve color="#1f77b4" name="/iiwa_front/joint_states/F_joint_1/effort"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot flip_y="false" flip_x="false" style="Lines" mode="TimeSeries">
        <range right="1656161145.674849" bottom="-21.110772" top="-20.794802" left="1656161130.674926"/>
        <limitY/>
        <curve color="#1f77b4" name="/iiwa_front/joint_states/F_joint_2/effort"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot flip_y="false" flip_x="false" style="Lines" mode="TimeSeries">
        <range right="1656161145.674849" bottom="0.395243" top="0.506749" left="1656161130.674926"/>
        <limitY/>
        <curve color="#1f77b4" name="/iiwa_front/joint_states/F_joint_3/effort"/>
       </plot>
      </DockArea>
     </DockSplitter>
     <DockSplitter count="4" orientation="|" sizes="0.25;0.25;0.25;0.25">
      <DockArea name="...">
       <plot flip_y="false" flip_x="false" style="Lines" mode="TimeSeries">
        <range right="1656161145.674849" bottom="23.000516" top="23.322329" left="1656161130.674926"/>
        <limitY/>
        <curve color="#1f77b4" name="/iiwa_front/joint_states/F_joint_4/effort"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot flip_y="false" flip_x="false" style="Lines" mode="TimeSeries">
        <range right="1656161145.674849" bottom="-1.145280" top="-1.112002" left="1656161130.674926"/>
        <limitY/>
        <curve color="#1f77b4" name="/iiwa_front/joint_states/F_joint_5/effort"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot flip_y="false" flip_x="false" style="Lines" mode="TimeSeries">
        <range right="1656161145.674849" bottom="-0.672193" top="-0.657474" left="1656161130.674926"/>
        <limitY/>
        <curve color="#1f77b4" name="/iiwa_front/joint_states/F_joint_6/effort"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot flip_y="false" flip_x="false" style="Lines" mode="TimeSeries">
        <range right="1656161145.674849" bottom="0.004414" top="0.015277" left="1656161130.674926"/>
        <limitY/>
        <curve color="#1f77b4" name="/iiwa_front/joint_states/F_joint_7/effort"/>
       </plot>
      </DockArea>
     </DockSplitter>
    </DockSplitter>
   </Container>
  </Tab>
  <currentTabIndex index="2"/>
 </tabbed_widget>
 <use_relative_time_offset enabled="0"/>
 <!-- - - - - - - - - - - - - - - -->
 <!-- - - - - - - - - - - - - - - -->
 <Plugins>
  <plugin ID="DataLoad CSV">
   <default delimiter="0" time_axis=""/>
  </plugin>
  <plugin ID="DataLoad ROS bags">
   <use_header_stamp value="false"/>
   <discard_large_arrays value="true"/>
   <max_array_size value="100"/>
   <boolean_strings_to_number value="true"/>
   <remove_suffix_from_strings value="true"/>
  </plugin>
  <plugin ID="DataLoad ULog"/>
  <plugin ID="ROS Topic Subscriber">
   <use_header_stamp value="false"/>
   <discard_large_arrays value="true"/>
   <max_array_size value="100"/>
   <boolean_strings_to_number value="true"/>
   <remove_suffix_from_strings value="true"/>
  </plugin>
  <plugin ID="UDP Server"/>
  <plugin ID="WebSocket Server"/>
  <plugin ID="Fast Fourier Transform"/>
  <plugin ID="Quaternion to RPY"/>
  <plugin ID="Reactive Script Editor">
   <library code="--[[ Helper function to create a ScatterXY series from arrays.&#xa;&#xa; series_name: name of the created ScatterXY series&#xa; prefix:      prefix of the timeseries, before the index of the array&#xa; suffix_X:   suffix to complete the name of the series containing the X value.&#xa;             if [nil], the index of the array will be used.&#xa; suffix_Y:   suffix to complete the name of the series containing the Y value&#xa; timestamp:   usually the tracker_time variable&#xa;              &#xa; Example:&#xa; &#xa; Assuming we have multiple series in the form:&#xa; &#xa;   /trajectory/node.{N}/position/x&#xa;   /trajectory/node.{N}/position/y&#xa;   &#xa; where {N} is the index of the array (integer). We can create a reactive series from the array with:&#xa; &#xa;   CreateSeriesFromArray( &quot;my_trajectory&quot;, &quot;/trajectory/node&quot;,  &quot;position/x&quot;, &quot;position/y&quot;, tracker_time );&#xa;]]--&#xa;&#xa;function CreateSeriesFromArray( series_name, prefix, suffix_X, suffix_Y, timestamp )&#xa;  --- create a new series or overwite the previous one&#xa;  new_series = MutableScatterXY.new(series_name)&#xa;  &#xa;  --- Append points to new_series&#xa;  index = 0&#xa;  while(true) do&#xa;&#xa;    x = index;&#xa;    -- if not nil, get the X coordinate from a series&#xa;    if suffix_X ~= nil then &#xa;      series_x = TimeseriesView.find( string.format( &quot;%s.%d/%s&quot;, prefix, index, suffix_X) )&#xa;      if series_x == nil then break end&#xa;      x = series_x:atTime(timestamp)&#x9; &#xa;    end&#xa;    &#xa;    series_y = TimeseriesView.find( string.format( &quot;%s.%d/%s&quot;, prefix, index, suffix_Y) )&#xa;    if series_x == nil then break end &#xa;    y = series_y:atTime(timestamp)&#xa;    &#xa;    new_series:push_back(x,y)&#xa;    index = index+1&#xa;  end&#xa;end&#xa;"/>
   <scripts/>
  </plugin>
  <plugin ID="CSV Exporter"/>
  <plugin ID="ROS /rosout Visualization"/>
  <plugin ID="ROS Topic Re-Publisher"/>
 </Plugins>
 <!-- - - - - - - - - - - - - - - -->
 <previouslyLoaded_Datafiles/>
 <previouslyLoaded_Streamer name="ROS Topic Subscriber"/>
 <!-- - - - - - - - - - - - - - - -->
 <customMathEquations/>
 <snippets/>
 <!-- - - - - - - - - - - - - - - -->
</root>

